src/ConditionAnalyzer/Analyzer.php

```diff
@@ -202,7 +210,7 @@ private function hasValidQueryOperator($hash, $range = null)
         $hashConditionType = $this->getCondition($hash)['type'] ?? null;
         $validQueryOp = ComparisonOperator::isValidQueryDynamoDbOperator($hashConditionType);
 
-        if ($validQueryOp && $range) {
+        if ($validQueryOp && $range && $this->getCondition($range)) {
```

Comment:
is this needed? should the check be in line 214 instead?

Reply:
> 

No, actually. It needs to be there. I tested locally. The ranged condition is valid to be Query when it is null.

So, this should be `&& $this->getCondition($range) !== null` instead